std/range/package.d

```diff
@@ -2553,6 +2557,124 @@ auto takeNone(R)(R range)
 }
 
 /++
+ + Return a _range advanced to within $(D n) elements of the end of
+ + $(D _range).
+ +
+ + Intended as the _range equivalent of the Unix
+ + $(WEB en.wikipedia.org/wiki/Tail_%28Unix%29, _tail) utility. When the length
+ + of $(D _range) is less than or equal to $(D n), $(D _range) is returned
+ + as-is.
+ +
+ + Completes in $(BIGOH 1) steps for ranges that support slicing and have
+ + length. Completes in $(BIGOH _range.length) time for all other ranges.
+ +/
+Range tail(Range)(Range range, size_t n)
+    if (isForwardRange!Range && !isInfinite!Range)
+{
+    static if (hasLength!Range)
+    {
+        // Evaluates `length` fewer times than calling `drop` would in the
+        // slicing case
+        immutable length = range.length;
+
+        if (n >= length)
+            return range;
+
+        static if (hasSlicing!Range)
+            return range[length - n .. length];
+        else
+            return .drop(range, length - n);
+    }
+    else
+    {
+        if (range.empty)
```

Comment:
It means more calls to `save` when the length of `range` is `0` or `n + 1`, but I suppose there's no reason to optimize for these cases.